iceoryx_posh/test/integrationtests/test_shm_sigbus_handler.cpp

```diff
@@ -38,11 +38,7 @@ TEST(ShmCreatorDeathTest, AllocatingTooMuchMemoryLeadsToExitWithSIGBUS)
         TEST_SHM_NAME, iox::posix::AccessMode::readWrite, iox::posix::OwnerShip::mine);
     badShmProvider.addMemoryBlock(&badmempools);
 
-    EXPECT_DEATH(badShmProvider.create(),
-                 "\033\\[0;1;97;41mFatal error:\033\\[m the available memory is insufficient. Cannot allocate mempools "
```

Comment:
This means that the Sigbus handler is not called anymore. Instead the following comes up:
```
[  DEATH   ] Reserving 107374194976 bytes in the shared memory [/test_name]
[  DEATH   ] 
SharedMemory still there, doing an unlink of /test_name
Reserving 3328 bytes in the shared memory [/test_name]
[ Reserving shared memory successful ] 
2021-02-17 12:08:02.606 [ Debug ]: Registered memory segment 0x7fa7f545c000 with size 3328 to id 1
```
We should inform the user with a message that the size is not sufficient for creating the shared memory.

See comment above regarding sigbus handler.

What you can do to not compare against the complete error text with all these special character is that you use regex to compare against a unique part of the error text.
```cpp
    EXPECT_DEATH(badShmProvider.create(), ".*available memory is insufficient. Cannot allocate mempools in shared "
        "memory. Please make sure that enough memory is available.*");
```

Reply:
The user is getting exactly this message: 
```
mmap failed:
One or more of the following failures happened:
  1. Not enough memory available.
  2. The maximum supported number of mappings is exceeded.
  3. Partial unmapping of an already mapped memory region dividing it into two parts.
  4. The processes maximum size of data segments is exceeded/
  5. The sum of the number of pages used for length and the pages used for offset would overflow and unsigned long. (only 32-bit architecture)

Unable to create shared memory with the following properties [ name = /roudi, access mode = AccessMode::ReadWrite, ownership = OwnerShip::Mine, mode = 010101, sizeInBytes = 13219012391 ]
```

Reply:
It is implemented in shared_memory_object!